src/pint/residuals.py

```diff
@@ -679,6 +691,13 @@ def data_error(self):
             errors.append((rs.residual_type, rs.get_data_error()))
         return collections.OrderedDict(errors)
 
+    def calc_chi2(self):
```

Comment:
I think this should call `calc_chi2` on the components - that is, it should not use the cache. The point of `calc_chi2` is to force recomputation, and this implementation doesn't.

Alternatively, you could have CombinedResiduals not cache anything, just compute the sum of the chi2 attributes of the parts every time it is requested. This accomplishes the same thing (and is what is there now).

Reply:
Yeah, you're right -- the alternatively part.  I was messing with something more involved for this calc_chi2 (calling WidebandTOAFitter in some cases), but turns out it didn't do what I wanted.  So I can revert.